sbt-app/src/sbt-test/actions/compile-time-only/build.sbt

```diff
@@ -5,7 +5,7 @@ lazy val expectErrorNotCrash = taskKey[Unit]("Ensures that sbt properly set type
 
 expectErrorNotCrash := {
   val fail = (Compile / compileIncremental).failure.value
-  fail.directCause match
+  Incomplete.allExceptions(fail).headOption match
```

Comment:
The `Def.taskIf` wrapping introduces an indirection ... I assume it's OK as user shouldn't rely on the exact hierarchy since plugins may change it anyway.